message_ix/core.py

```diff
@@ -107,3 +116,73 @@ def vintage_and_active_years(self):
         year_pairs = [(y_v, y_a) for y_v, y_a in combinations if y_v <= y_a]
         v_years, a_years = zip(*year_pairs)
         return v_years, a_years
+
+    def solve(self, **kwargs):
+        """Solve a Message Scenario. See ixmp.Scenario.solve() for arguments"""
+        return super(Scenario, self).solve(model='MESSAGE', **kwargs)
+
+    def clone(self, model=None, scen=None, annotation=None, keep_sol=True,
+              first_model_year=None):
+        """clone the current scenario and return the new scenario
+
+        Parameters
+        ----------
+        model : string
+            new model name
+        scen : string
+            new scenario name
+        annotation : string
+            explanatory comment (optional)
+        keep_sol : boolean, default: True
+            indicator whether to include an existing solution
+            in the cloned scenario
+        first_model_year: int, default None
+            new first model year in cloned scenario
+            ('slicing', only available for MESSAGE-scheme scenarios)
+        """
+        self._keep_sol = keep_sol
+        self._first_model_year = first_model_year or 0
+        model = self.model if not model else model
+        scen = self.scenario if not scen else scen
+        return Scenario(self.platform, model, scen, annotation=annotation,
+                        cache=self._cache, clone=self)
+
+    def rename_technology(self, from_tech, to_tech):
+        """Rename a technology in a scenario
+
+        Parameters
+        ----------
+        from_tech : str
+            original technology name
+        to_tech : str
+            new technology name
+        """
+        self.check_out()
+
+        # search for from_tech in sets and replace
+        for item in self.set_list():
+            value = self.set(item)
+            if isinstance(value, pd.DataFrame):
+                if 'technology' in value.columns and not value.empty:
+                    value = value[value['technology'] == from_tech]
+                    if not value.empty:
+                        value['technology'] = to_tech
+                        self.add_set(item, value)
+            elif value.isin([from_tech]).any():  # value is pd.Series
+                self.add_set(item, to_tech)
+
+        # search for from_tech in pars and replace
+        for item in self.par_list():
+            if 'technology' not in self.idx_names(item):
+                continue
+            df = self.par(item, filters={'technology': from_tech})
+            if not df.empty:
+                df['technology'] = to_tech
+                self.add_par(item, df)
+
+        # this removes all instances of from_tech in the model
+        self.remove_set('technology', from_tech)
+
+        # recommit
```

Comment:
why `recommit`?

Reply:
Sorry, is this question for why commit at all, or why the comment says recommit?

Reply:
just the docstring